### PR TITLE
Add org README

### PR DESCRIPTION
**Pull request by tcharding**
Add a readme file in `profile/` which will be displayed on the https://github.com/rust-bitcoin page.

Add content describing and linking to all the crates and repositories in the org.

Inspired by: https://github.com/rust-bitcoin/rust-bitcoin/issues/2656
Review the rendered format: https://github.com/tcharding/rust-bitcoin-org.github/blob/04-05-readme/profile/README.md

I don't know exactly how this will be rendered on the httsp://github.com/rust-bitcoin page, I only just learned this is possible.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,5 @@
-# .github
\ No newline at end of file
+# .github
+
+This repository is special, its sole purpose is to contain a
```
Comment (apoelstra) [inline README.md:3]: `,` should be `.`.

File: profile/README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+If you write proprietary software and want better integration with these crates by all means raise
+issues on the respective repository. We are interested in all users and you usecase may help us
+improve the libraries, however all development is done in the open and will stay that way.
+
+## Other Crates
+
+Other crates/repositories hosted here and developed and maintained by the community.
+
+These crates are developed and maintained by a subset of the rust-bitcoin devs or other members of
+the community. We are quite open with hosting crates in the org, as such the bar for entry is only:
```
Comment (apoelstra) [inline profile/README.md:41]: I would soften this language a bit.

```
We are fairly open about hosting any bitcoin-related crates in this org, provided that they are actively maintained and:
```

If we say "quite open" and use the word "only" I worry that we'll get requests for questionable things and then have dumb fights about "well your README says you'll take it". (OTOH it is true that we're pretty open.)

File: profile/README.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+- Repository contains Bitcoin specific crates (implies Bitcoin Only and not general purpose crates)
+- Repository has at least minimal CI
+- Author has made at least some effort to adhere to practices and standards maintained in the core crates.
+
+As such any of these crates should be personally reviewed by projects before use, including
+reviewing new versions and not just blindly upgrading - you have been warned.
```
Comment (apoelstra) [inline profile/README.md:48]:
```
These crates may not be reviewed or maintained by the primary rust-bitcoin maintainers. Keep this in mind when choosing to use them as dependencies. And as with any dependencies, you should review new versions rather than blindly upgrading.
```

Comment (apoelstra) [reply profile/README.md:48]: We don't want to imply that these crates are *bad* (otherwise we really would move them out of the org). Just that they have a different trust profile than the main crates.

**Comment by tcharding** [comment]
Used all suggestions, thanks man.

**Review by apoelstra — APPROVED** [review]
utACK b36cbf6c0ed73a29edcf642ea65334365a9b8083

**Comment by tcharding** [comment]
@storopoli would you mind taking a look at this one please when you have a minute?

File: profile/README.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
+# Rust Bitcoin Community
+
+A Series of projects to implement various Bitcoin protocols in Rust.
+
+## Core Crates
+
+The "core" crates developed and maintained collectively by the `rust-bitcoin` devs.
+
+For these crates we hold ourselves to high standards relating to security, testing, release
+schedule, API design, and coding style. Our current focus is on stabilising these crates [0].
```
Comment (storopoli) [inline profile/README.md:10]: This `[0]` footnote is broken.

```suggestion
schedule, API design, and coding style. Our current focus is on stabilising these crates[^0].
```

Ref: https://github.blog/changelog/2021-09-30-footnotes-now-supported-in-markdown-fields/

File: profile/README.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+these crates to follow software engineering best practices e.g., semver compliance.
+
+If we fall short of these stated aims, or can in anyway improve our support of Bitcoin free software
+projects, please raise issues or contact maintainers directly.
+
+If you write proprietary software and want better integration with these crates by all means raise
+issues on the respective repository. We are interested in all users and your usecase may help us
+improve the libraries, however all development is done in the open and will stay that way.
+
+[0]  (Please note `internals` will explicitly never stabalize.)
```
Comment (storopoli) [inline profile/README.md:36]: Typo and broken footnote

```suggestion
[^0]: Please note `internals` will explicitly never stabilize.
```

File: profile/README.md (hunk 4 above)
Comment (storopoli) [reply profile/README.md:10]: Also down below we say `stabilize`, so maybe we should also use the american "way" here:

```suggestion
schedule, API design, and coding style. Our current focus is on stabilizing these crates [0].
```

File: profile/README.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
+# Rust Bitcoin Community
+
+A Series of projects to implement various Bitcoin protocols in Rust.
+
```
Comment (storopoli) [inline profile/README.md:4]:
```suggestion

## TLDR

`rust-bitcoin` hosts the [`bitcoin`](https://crates.io/crates/bitcoin) crate that
supports the Bitcoin network protocol and associated primitives.
It is designed for Rust programs built to work with the Bitcoin network.

```

Something like this. Feel free to edit or add more to it.

The idea is that if a newcomer sees the profile README he gets a nice "elevator pitch" of what `rust-bitcoin` is all about without having "death by analysis" by having to go into a bunch of different crates' READMEs.

I would have much appreciated something like this when I was starting to play around with `rust-bitcoin` a while ago.

Feel free to ignore this if you find that my arguments are invalid.

File: profile/README.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+io | [bitcoin-io](https://crates.io/crates/bitcoin-io) | [rust-bitcoin](https://github.com/rust-bitcoin/rust-bitcoin)
+ordered | [ordered](https://crates.io/crates/ordered) | [rust-ordered](https://github.com/rust-bitcoin/rust-ordered)
+secp256k1 | [secp256k1](https://crates.io/crates/secp256k1) | [rust-secp256k1](https://github.com/rust-bitcoin/rust-secp256k1)
+units | [bitcoin-units](https://crates.io/crates/bitcoin-units) | [rust-bitcoin](https://github.com/rust-bitcoin/rust-bitcoin)
+
+We aim to make usage of these crates secure, ergonomic, and intuitive. We try to enable users of
+these crates to follow software engineering best practices e.g., semver compliance.
+
+If we fall short of these stated aims, or can in anyway improve our support of Bitcoin free software
+projects, please raise issues or contact maintainers directly.
```
Comment (storopoli) [inline profile/README.md:30]: We could link to the https://github.com/rust-bitcoin/rust-bitcoin/blob/master/SECURITY.md

And also we could do something like enumerating the maintainers, as in
https://github.com/rust-bitcoin/rust-bitcoin/blob/47527abd536cbc30e47bce127cc9d2b7c8cc0b98/CONTRIBUTING.md?plain=1#L142-L151
I know that there's a risk this gets outdated...

File: profile/README.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+improve the libraries, however all development is done in the open and will stay that way.
+
+[0]  (Please note `internals` will explicitly never stabalize.)
+
+## Other Crates
+
+Other crates/repositories hosted here and developed and maintained by the community.
+
+These crates are developed and maintained by a subset of the `rust-bitcoin` devs or other members of
+the community. We are fairly open about hosting any bitcoin-related crates in this org, provided
```
Comment (storopoli) [inline profile/README.md:43]:
```suggestion
the community. We are fairly open about hosting any Bitcoin-related crates in this organization, provided
```

Bitcoin is the protocol/software, bitcoin is any monetary amount in BTC.

File: profile/README.md [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+that they are actively maintained and:
+
+- Repository contains Bitcoin specific crates (implies Bitcoin Only and not general purpose crates)
```
Comment (storopoli) [inline profile/README.md:46]:
```suggestion
- contains Bitcoin-specific crates (implies Bitcoin-only and not general purpose crates)
```

File: profile/README.md [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+- Repository has at least minimal CI
```
Comment (storopoli) [inline profile/README.md:47]:
```suggestion
- has at least minimal CI
```

File: profile/README.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+- Author has made at least some effort to adhere to practices and standards maintained in the core
```
Comment (storopoli) [inline profile/README.md:48]:
```suggestion
- author has made at least some effort to adhere to practices and standards maintained in the core
```

File: profile/README.md [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+Repositories that are not crates (libraries).
+
+Name | Repository | Description
+---|---|---
+maintainer-tools | [rust-bitcoin-maintainer-tools](https://github.com/rust-bitcoin/rust-bitcoin-maintainer-tools) | Scripts, test vectors, and other things used by or across multiple repositories in the `rust-bitcoin` ecosystem.
+website | [rust-bitcoin.github.io](https://github.com/rust-bitcoin/rust-bitcoin.github.io) | Source for the org website https://rust-bitcoin.org
+workshop | [workshop](https://github.com/rust-bitcoin/) | Conference presentation material
+.github | [.github](https://github.com/rust-bitcoin/) | Hosts this readme file
+
+## Archived (moved or unmaintained)
```
Comment (storopoli) [inline profile/README.md:77]:
```suggestion
## Archived (moved or unmaintained) crates
```

Also we should add a brief explainer, as all preceding H2, before the table.

File: profile/README.md [hunk 13]
```diff
@@ -0,0 +1,90 @@
+# Rust Bitcoin Community
+
+A Series of projects to implement various Bitcoin protocols in Rust.
+
+## Core Crates
+
+The "core" crates developed and maintained collectively by the `rust-bitcoin` devs.
```
Comment (storopoli) [inline profile/README.md:7]: Same as below with the "maintainers" argument.
Where do we find about the `rust-bitcoin devs`?

Either link to something or list them?

**Review by storopoli — COMMENTED** [review]
I am still missing a TLDR.

I think we should start with that instead.
This is something quite valuable for beginners and newcomers,
and experienced/well-acquainted can easily skip.

I've added a suggestion in the beginning of the file.

Finally, in the footnote we say refer to the crate `internals` in backticks, aka code. I think that this is the optimal way to refer to crates, and would like to see the backtick also applied to the others, e.g. `base58`...

File: profile/README.md (hunk 6 above)
Comment (tcharding) [reply profile/README.md:4]: Good suggestion, I'll work this in.

**Comment by tcharding** [comment]
Thanks for the review man, I'll work in your suggestions.

File: profile/README.md (hunk 4 above)
Comment (apoelstra) [reply profile/README.md:10]: We also have the spelling `stabalize` (which @tcharding has very consistently used so I assume it's a dialect thing). I think we should standardize on `stabilize` and `stabilizing`.

Comment (tcharding) [reply profile/README.md:10]: yeah I have my editor set to UK English and manually use US English when working, perhaps I have to bight the bullet and just use US English in all my life - but that hurts a bit.

Comment (apoelstra) [reply profile/README.md:10]: Yeah, as a Canadian I feel you. I choose instead to fill my brain with Americanisms and then do them all by memory (and at work we employ Americans to further correct my grammar).

File: profile/README.md (hunk 6 above)
Comment (tcharding) [reply profile/README.md:4]: Used as is (with column width 100).

File: profile/README.md (hunk 7 above)
Comment (tcharding) [reply profile/README.md:30]: Added security link. The maintainer list is crate specific, I'd prefer not to add any names to the org readme. @apoelstra please weigh in if you think differently.

File: profile/README.md (hunk 8 above)
Comment (tcharding) [reply profile/README.md:43]: Used both suggestions (and remove a further "org" below). Thanks

File: profile/README.md (hunk 9 above)
Comment (tcharding) [reply profile/README.md:46]: Done!

File: profile/README.md (hunk 10 above)
Comment (tcharding) [reply profile/README.md:47]: I think the original is correct, what makes you think we should remove "Repository"?

File: profile/README.md (hunk 11 above)
Comment (tcharding) [reply profile/README.md:48]: Are you saying all three list items should use lower case starting character? I went the other way and added full stops and used capital. Do say if you think this is wrong - happy to learn.

File: profile/README.md (hunk 12 above)
Comment (tcharding) [reply profile/README.md:77]: Added "These crates have either been moved to another location or are no longer actively maintained."

File: profile/README.md (hunk 13 above)
Comment (tcharding) [reply profile/README.md:7]: I'm not personally super comfortable defining this group, I think the git index has all the info needed.

File: profile/README.md (hunk 10 above)
Comment (storopoli) [reply profile/README.md]: It's more aesthetic. The original is correct, just that there is a repetition in

- Repository this
- Repository that

We could be more succinct and say:

- this
- that

Comment (storopoli) [reply profile/README.md]: It's bike-shedding, feel free to ignore...

File: profile/README.md (hunk 11 above)
Comment (storopoli) [reply profile/README.md:55]: I honestly don't know. But my impression is that the colon, semicolons, and commas they don't end a sentence.
That's the role of the period.
For example, if you are saying something and then you hit a semicolon, you don't start the next word capitalized, right?

The same with the colon.
In a way this

> My things:
>
> - thing one
> - thing two

Is generally written in more formal documents as:

> My things:
>
> - thing one; and
> - thing two.

With semicolons and periods to end.

So this can also be written as:

> My things: thing one; and thing two.

But I don't know if I am correct. Someone or some book taught me this a very long time ago. Can't recollect any details.

File: profile/README.md (hunk 13 above)
Comment (storopoli) [reply profile/README.md:13]: Then, we can specify somehow that these "devs" are the ones that appear in the commit graph?

File: profile/README.md (hunk 11 above)
Comment (tcharding) [reply profile/README.md]: Cool, I read one web page and updated with my new found knoweldge.

ref: https://editorsmanual.com/articles/lists/

File: profile/README.md (hunk 13 above)
Comment (tcharding) [reply profile/README.md]: I've added
```
The "core" crates developed and maintained collectively by the `rust-bitcoin` devs. There is a list
of official [maintainers](https://github.com/rust-bitcoin/rust-bitcoin/blob/master/CONTRIBUTING.md#repository-maintainers)
for the `bitcoin` crate, for other crates please see the git index to get an idea of who is who.
```

File: profile/README.md (hunk 7 above)
Comment (apoelstra) [reply profile/README.md]: Yeah, agreed, let's leave names out of this doc for now.

**Comment by tcharding** [comment]
Can you ack and approve please @storopoli

**Review by storopoli — APPROVED** [review]
ACK d9ac89075c43306519d96c9ef18975ff78e29318

**Comment by tcharding** [comment]
We forgot about this, I merged it while investigating https://github.com/rust-bitcoin/rust-bitcoin/issues/2656. Check it out: https://github.com/rust-bitcoin

cc @apoelstra
